### PR TITLE
Update versions to `5.1-alpha1-SNAPSHOT` matching httpcore 5.1.x

### PR DESCRIPTION
**Pull request by carterkozak**
This is a prerequisite to passing through the ResponseOutOfOrderStrategy for HTTPCLIENT-2104.

It looks like the update exposes some regressions in httpcore-5.1.x.

### CODE REVIEW

**Comment by ok2c** [comment]
@carterkozak Generally HC projects do not have SNAPSHOT dependencies for various, mostly historic, reasons. Once HttpCore is ready I will cut an official ALPHA or BETA release.

File: pom.xml [hunk 1]
```diff
@@ -61,13 +61,13 @@
...
     <url>https://github.com/apache/httpcomponents-client/tree/${project.scm.tag}</url>
-    <tag>5.0.2-SNAPSHOT</tag>
+    <tag>5.1.0-SNAPSHOT</tag>
   </scm>
 
   <properties>
     <maven.compiler.source>1.7</maven.compiler.source>
     <maven.compiler.target>1.7</maven.compiler.target>
-    <httpcore.version>5.0.1</httpcore.version>
+    <httpcore.version>5.1-alpha1-SNAPSHOT</httpcore.version>
```
Comment (michael-o) [inline pom.xml:70]: Indeed, my bad. Snapshots should not appear in deps.

**Comment by carterkozak** [comment]
Sounds good, I can revert the dependency on httpcore snapshots. I'm not sure if the test failures are intentional behavior changes, do you have any idea?

**Comment by ok2c** [comment]
@carterkozak @michael-o Can we focus on resolving HTTPCORE-640? Once resolved I will cut a proper ALPHA / BETA.

**Comment by carterkozak** [comment]
@ok2c Of course, I've opened a PR on -core with a fix for the flaking test. There's definitely a behavior change introduced in jre13, but I think the alogrithm is still reasonable without changes.

I've updated this PR reverting the snapshot dependency.

**Pull request closed without merging**